server/jdkhttp-server/src/main/scala/sttp/tapir/server/jdkhttp/internal/CircularBuffer.scala

```diff
@@ -0,0 +1,46 @@
+package sttp.tapir.server.jdkhttp.internal
+
+import scala.collection.IndexedSeqView
+
+class CircularBuffer(bufferSize: Int) {
```

Comment:
Not super sure if this needed to be a class or if i should just do something inline. Or even if i should use the circular buffer for everything like i did.